Remove `java_bytecode_parse_treet::classt::swap` and return using `optionalt` instead.

thomasspriggs opened the pull request:
This removes the `swap` method as suggested in https://github.com/diffblue/cbmc/pull/2011
Avoiding having a `swap`method, makes updates less error prone.

thomasspriggs committed:
Remove `java_bytecode::swap` and return using `optionalt` instead.

This removes the `swap` method as suggested in https://github.com/diffblue/cbmc/pull/2011
Avoiding having a `swap`method, makes updates less error prone.

peterschrammel reviewed (commented):
Looks good. Please describe in the body of the commit message why you made this change. I suggest squashing the two commits.

tautschnig commented:
#2011 is now merged, in case this matters. 

Edit: it does, a rebase is now needed.

thomasspriggs commented:
I have added a commit message and squashed the two commits are requested by Peter Schrammel. I have performed the required rebase as noted by Michael Tautschnig.

kroening commented:
May I suggest to add
`java_bytecode_parse_treet(const java_bytecode_parse_treet &) = delete;
java_bytecode_parse_treet &operator=(const java_bytecode_parse_treet &) = delete;`
to prevent accidental copying, given that these parse trees might be large.

thomasspriggs committed:
Delete copy constructor of `class java_bytecode_parse_treet`.

This deletes the default copy constructor of
`java_bytecode_parse_treet`, but allows move construction.
This enforces allowing move semantics only and prevents accidental
copying.

thomasspriggs committed:
Fix instances where copying was being used instead of moving.

thomasspriggs committed:
Switch `push_back` to `emplace_back` when constructing `parse_trees`.
